aiida_mlip/calculations/singlepoint.py

```diff
@@ -0,0 +1,226 @@
+        spec.input("precision", valid_type=Str, help="Precision level for calculation")
+        spec.input(
+            "device",
+            valid_type=Str,
+            required=False,
+            default=lambda: Str("cpu"),
+            help="Device in which to run calculation(cpu, gpu...)",
+        )
+
+        spec.input(
+            "xyzoutput",
+            valid_type=Str,
+            required=False,
+            default=lambda: Str(cls._XYZ_OUTPUT),
+            help="Name of the xyz output file",
+        )
+
+        spec.input(
+            "metadata.options.input_filename",
+            valid_type=str,
+            default=cls._DEFAULT_INPUT_FILE,
+        )
+        spec.input(
+            "metadata.options.output_filename",
+            valid_type=str,
+            default=cls._DEFAULT_OUTPUT_FILE,
+        )
+
+        spec.input(
+            "metadata.options.scheduler_stdout",
+            valid_type=str,
+            default="_scheduler-stdout.txt",
+            help="Filename to which the content of stdout of the scheduler is written.",
+        )
+        spec.inputs["metadata"]["options"]["parser_name"].default = "janus.parser"
+
+        # Define outputs. The default output is a dictionary with the content of the xyz file as a dict
+        spec.output(
+            "results_dict",
+            valid_type=Dict,
+            help="The `results_dict` output node of the successful calculation.",
+        )
+        print("defining outputs")
+        spec.output("log_output", valid_type=SinglefileData)
+        spec.output("xyz_output", valid_type=SinglefileData)
+        print("defining outputnode")
+        spec.default_output_node = "results_dict"
+
+        # Exit codes
+        spec.exit_code(300, "INPUT_ERROR", message="Some problems reading the input")
+
+        spec.exit_code(
+            340,
+            "ERROR_OUT_OF_WALLTIME_INTERRUPTED",
+            message="The calculation stopped prematurely because it ran out of walltime",
+        )
+        spec.exit_code(
+            305,
+            "ERROR_MISSING_OUTPUT_FILES",
+            message="Some output files missing or cannot be read",
+        )
+        spec.exit_code(
+            306,
+            "ERROR_EMPTY_OUTPUT_FILES",
+            message="File.xyz is empty",
+        )
+        spec.exit_code(
+            307,
+            "ERROR_CONTENT_OUTPUT_FILES",
+            message="The output file does not contain the right content",
+        )
+
+    @classmethod
+    def validate_inputs(cls, value: dict) -> str:  # Need to work on this
+        """Check if the inputs are valid.
+
+        Parameters
+        ----------
+        value : dict
+            The inputs dictionary.
+
+        Returns
+        -------
+        str or None
+            Error message if validation fails, None otherwise.
+        """
+
+        # Check if  structure is given as it is required
+        for key in "structure":
```

Comment:
Does this want to be a string?
```suggestion
for key in structure:
```